### PR TITLE
Improve IP filtering logic

### PR DESCRIPTION
**Pull request by lgarvey**
This PR makes the IP white listing code aware of the correct position the client's IP should appear in the x-forwarded-for header.

### CODE REVIEW

File: config/settings/__init__.py [hunk 1]
```diff
@@ -38,9 +38,8 @@
 # to the external url, e.g: ALLOWED_HOSTS=invest.great.uat.uktrade.io
 ALLOWED_HOSTS = ['*']
 
-RESTRICT_ADMIN = os.environ.get('RESTRICT_ADMIN', False)
-ALLOWED_ADMIN_IPS = os.environ.get('ALLOWED_ADMIN_IPS', [])
-ALLOWED_ADMIN_IP_RANGES = os.environ.get('ALLOWED_ADMIN_IP_RANGES', [])
+RESTRICT_ADMIN = os.environ.get('RESTRICT_ADMIN', 'False') == 'True'
+ALLOWED_ADMIN_IPS = [ip.strip() for ip in os.environ.get('ALLOWED_ADMIN_IPS', '').split(',')]  # noqa: E501
```
Comment (michalc) [inline config/settings/__init__.py]: If `ALLOWED_ADMIN_IPS` isn't set, does this result in a list of `['']`? Better to just be an empty list? (Slightly paranoid that "somehow" an IP address of the empty string could make it through X-Forwarded-For...)

File: core/middleware.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+logger = logging.getLogger(__name__)
+
+
+def AdminIpRestrictionMiddleware(get_response):
+
+    def middleware(request):
+        if resolve(request.path).app_name == 'admin':
+            if settings.RESTRICT_ADMIN:
+                try:
+                    remote_address = request.META['HTTP_X_FORWARDED_FOR'].split(',')[-2].strip()  # noqa: E501
```
Comment (michalc) [inline core/middleware.py]: Can this also throw a key error if `HTTP_X_FORWARDED_FOR` isn't in the request? Should be handled?

File: core/middleware.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+logger = logging.getLogger(__name__)
+
+
+def AdminIpRestrictionMiddleware(get_response):
+
+    def middleware(request):
+        if resolve(request.path).app_name == 'admin':
+            if settings.RESTRICT_ADMIN:
```
Comment (michalc) [inline core/middleware.py]: Is there a use case where this isn't set but the middleware is used? Simpler to control restriction by using/not using the middleware?

File: config/settings/__init__.py (hunk 1 above)
Comment (lgarvey) [reply config/settings/__init__.py]: Yep, good point - will fix.

File: core/middleware.py (hunk 2 above)
Comment (lgarvey) [reply core/middleware.py]: Another good point. Will fix.

File: core/middleware.py (hunk 3 above)
Comment (michalc) [reply core/middleware.py]: ok!

**Pull request merged**